File: FitForge/Models/BodyMetric.swift
import Foundation
import SwiftData

/// 身体数据记录（体重、体脂、围度等）
@Model
final class BodyMetric {
    var id: UUID
    var date: Date
    var weightKg: Double?
    var bodyFatPercentage: Double?

    // 围度 (cm)
    var chestCm: Double?
    var waistCm: Double?
    var hipsCm: Double?
    var armCm: Double?
    var thighCm: Double?

    var notes: String

    init(
        date: Date = Date(),
        weightKg: Double? = nil,
        bodyFatPercentage: Double? = nil,
        chestCm: Double? = nil,
        waistCm: Double? = nil,
        hipsCm: Double? = nil,
        armCm: Double? = nil,
        thighCm: Double? = nil,
        notes: String = ""
    ) {
        self.id = UUID()
        self.date = date
        self.weightKg = weightKg
        self.bodyFatPercentage = bodyFatPercentage
        self.chestCm = chestCm
        self.waistCm = waistCm
        self.hipsCm = hipsCm
        self.armCm = armCm
        self.thighCm = thighCm
        self.notes = notes
    }
}

File: lib/models/body_metric.dart
/// 身体数据记录
class BodyMetric {
  BodyMetric({
    required this.id,
    DateTime? date,
    this.weightKg,
    this.bodyFatPercentage,
    this.chestCm,
    this.waistCm,
    this.hipsCm,
    this.armCm,
    this.thighCm,
  }) : date = date ?? DateTime.now();

  factory BodyMetric.fromJson(Map<String, dynamic> json) => BodyMetric(
    id: json['id'] as String,
    date: DateTime.parse(json['date'] as String),
    weightKg: (json['weightKg'] as num?)?.toDouble(),
    bodyFatPercentage: (json['bodyFatPercentage'] as num?)?.toDouble(),
    chestCm: (json['chestCm'] as num?)?.toDouble(),
    waistCm: (json['waistCm'] as num?)?.toDouble(),
    hipsCm: (json['hipsCm'] as num?)?.toDouble(),
    armCm: (json['armCm'] as num?)?.toDouble(),
    thighCm: (json['thighCm'] as num?)?.toDouble(),
  );
  final String id;
  final DateTime date;
  double? weightKg;
  double? bodyFatPercentage;
  double? chestCm;
  double? waistCm;
  double? hipsCm;
  double? armCm;
  double? thighCm;

  Map<String, dynamic> toJson() => {
    'id': id,
    'date': date.toIso8601String(),
    'weightKg': weightKg,
    'bodyFatPercentage': bodyFatPercentage,
    'chestCm': chestCm,
    'waistCm': waistCm,
    'hipsCm': hipsCm,
    'armCm': armCm,
    'thighCm': thighCm,
  };
}
